src/test/java/com/google/api/generator/gapic/composer/grpcrest/GrpcServiceCallableFactoryClassComposerTest.java

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.gapic.composer.grpcrest;
+
+import com.google.api.generator.engine.writer.JavaWriterVisitor;
+import com.google.api.generator.gapic.composer.grpc.GrpcServiceCallableFactoryClassComposer;
+import com.google.api.generator.gapic.model.GapicClass;
+import com.google.api.generator.gapic.model.GapicContext;
+import com.google.api.generator.gapic.model.Service;
+import com.google.api.generator.test.framework.Assert;
+import com.google.api.generator.test.framework.Utils;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.junit.Test;
+
+public class GrpcServiceCallableFactoryClassComposerTest {
+  @Test
+  public void generateServiceClasses() {
+    GapicContext context = GrpcRestTestProtoLoader.instance().parseShowcaseEcho();
+    Service echoProtoService = context.services().get(0);
+    GapicClass clazz =
+        GrpcServiceCallableFactoryClassComposer.instance().generate(context, echoProtoService);
+
+    JavaWriterVisitor visitor = new JavaWriterVisitor();
+    clazz.classDefinition().accept(visitor);
+    Utils.saveCodegenToFile(this.getClass(), "GrpcEchoCallableFactory.golden", visitor.write());
+    Path goldenFilePath =
+        Paths.get(Utils.getGoldenDir(this.getClass()), "GrpcEchoCallableFactory.golden");
+    Assert.assertCodeEquals(goldenFilePath, visitor.write());
```

Comment:
Consider checking the clazz.samples() as well. See similar [grpc tests](https://github.com/googleapis/gapic-generator-java/blob/main/src/test/java/com/google/api/generator/gapic/composer/grpc/ServiceClientClassComposerTest.java#L52).  See [Assert.assertGoldenSamples](https://github.com/googleapis/gapic-generator-java/blob/051713dec7706713900b8708e3a6b9ab0639438e/src/test/java/com/google/api/generator/test/framework/Assert.java#L68) and [assertEmptySamples](https://github.com/googleapis/gapic-generator-java/blob/051713dec7706713900b8708e3a6b9ab0639438e/src/test/java/com/google/api/generator/test/framework/Assert.java#L53). Even if there aren't any samples, I think it's good to be clear in the tests which classes generate samples and which don't.